respy/pre_processing/model_processing.py

```diff
@@ -1,105 +1,163 @@
 """Process model specification files or objects."""
+import collections
+import re
 import warnings
 from pathlib import Path
 
 import numpy as np
 import pandas as pd
 import yaml
+from estimagic.optimization.utilities import sdcorr_params_to_matrix
 
 from respy.config import DEFAULT_OPTIONS
 from respy.pre_processing.model_checking import _validate_options
 
 warnings.simplefilter("error", category=pd.errors.PerformanceWarning)
 
 
+def process_params_and_options(params, options):
```

Comment:
I think for convenience, we should support path-objects as inputs. I assume that you would like to store your model specification on disk in csv and yaml rather than pickle which is better for version control. Or do you always pickle you skillmodels models?

Reply:
Ok, then we keep the path input. But I still don't understand why we want to save processed options. Why not just generate them from the user options and params whenever they are needed?

Reply:
I thought a user might want to save the processed options as defaults may change which makes reproducing former results more difficult. But, the user does not see the processed options anyway. So, you are probably right. If it is okay with you, I'll delete save_option and related parts.

Reply:
Yes, I think the user should not see the processed options. After this round of major changes we should have a stable params/options interface.